### PR TITLE
Chef 15 compatibility fixes

### PR DESCRIPTION
**Pull request by detjensrobert**
Idempotency has been disabled for this cookbook as the `yum_version_lock` resource executes on every run.

### CODE REVIEW

**Comment by detjensrobert** [comment]
[Chef 15 compatibility fixes](https://trello.com/c/k2lpIE2U/1570-chef-15-compatibility-fixes)

**Comment by ramereth** [comment]
[Chef 15: osl-docker](https://trello.com/c/14JBn6uH/1756-chef-15-osl-docker)

File: attributes/default.rb [hunk 1]
```diff
@@ -26,6 +26,6 @@
   }
 default['osl-docker']['prune']['volume_filter'] = []
 default['osl-docker']['tls'] = false
-default['osl-docker']['host'] = node['osl-docker']['tls'] ? 'tcp://127.0.0.1:2376' : nil
+default['osl-docker']['host'] = node['osl-docker']['tls'] ? 'tcp://127.0.0.1:2376' : ''
```
Comment (ramereth) [inline attributes/default.rb:29]: Why was this changed?

Comment (detjensrobert) [reply attributes/default.rb:29]: The new `osl-shell` cannot have `nil` as an argument, and even though theres a `not_if` guard it still throws a validation error. I changed the default value attribute value and the nil checks to check for empty string instead.

Comment (ramereth) [reply attributes/default.rb:29]: So it looks like to properly work around this we should change ``osl_shell_environment[DOCKER_HOST]`` to be wrapped in an ``if`` block instead of using the ``not_if`` guard. That should work around the issue properly and not require changing this.

**Comment by detjensrobert** [comment]
!bump minor *
